### PR TITLE
[1.4] More tModPorter tests

### PR DESCRIPTION
**Pull request by direwolf420**
### Description
Added a few common test cases encountered in porting.
This PR does NOT add the rewriters!

Also let me know if the way I added comments is acceptable/correct, this is my first tModPorter related PR.

I also noticed inconsistencies regarding line endings (CRLF/LF/Mixed), but it does not affect me.

### CODE REVIEW

File: tModPorter/tModPorter.Tests/TestData/FontsTest.Expected.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using Terraria.ModLoader;
```
Comment (Chicken-Bones) [inline tModPorter/tModPorter.Tests/TestData/FontsTest.Expected.cs:1]: we'll probably leave the unnecessary `using Terraria;`, as it's easy for modders to cleanup later, and figuring out whether it's now redundant is a bit of a pain.

File: tModPorter/tModPorter.Tests/TestData/FontsTest.Expected.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+using ReLogic.Graphics;
+using Terraria.GameContent; //FontAssets
```
Comment (Chicken-Bones) [inline tModPorter/tModPorter.Tests/TestData/FontsTest.Expected.cs]: Do you think adding a comment about why usings were added is a good idea? Or should we just add them silently. What if they end up required by different rewriters for different reasons?

File: tModPorter/tModPorter.Tests/TestData/GlobalItemTest.Expected.cs [hunk 3]
```diff
@@ -1,7 +1,19 @@
 using Terraria;
 using Terraria.ModLoader;
+using Terraria.ModLoader.IO;
+using Microsoft.Xna.Framework;
```
Comment (Chicken-Bones) [inline tModPorter/tModPorter.Tests/TestData/GlobalItemTest.Expected.cs:4]: Additional framework import? Probably just a leftover.

File: tModPorter/tModPorter.Tests/TestData/ModProjectileTest.Expected.cs [hunk 4]
```diff
@@ -1,9 +1,17 @@
 using System;
 using Terraria.ModLoader;
+using Microsoft.Xna.Framework; // Vector2, Color
+// Blank
```
Comment (Chicken-Bones) [inline tModPorter/tModPorter.Tests/TestData/ModProjectileTest.Expected.cs]: Yeah as above, we probably won't remove redundant usings. Could be considered helpful in the future? But it's a standard feature of VS

File: tModPorter/tModPorter.Tests/TestData/SoundTest.Expected.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+		SoundEngine.PlaySound(SoundID.Zombie10, player.position);
+
+		SoundEngine.PlaySound(SoundID.Item10, player.position);
+		SoundEngine.PlaySound(SoundID.NPCHit10, player.position);
+		SoundEngine.PlaySound(SoundID.NPCDeath10, player.position);
+		SoundEngine.PlaySound(SoundID.Zombie10, player.position);
+
+		//TODO modded sound retrieval&playback
+		//TODO non-dedicated IDs (Compile_Error req?)
+		//TODO pitch + volume
```
Comment (Chicken-Bones) [inline tModPorter/tModPorter.Tests/TestData/SoundTest.Expected.cs]: These comments are great! It's good to be reminded why test cases exist. Just make sure the comments are in the same place in the input and expected `.cs` files.

File: tModPorter/tModPorter.Tests/TestData/SimpleRenamedVanillaMembersTest.Expected.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
+using Terraria;
+using Terraria.Chat; // ChatHelper
+using Terraria.ID;
+using Microsoft.Xna.Framework;
+
+public class SimpleRenamedVanillaMembersTest
+{
+	void Method() {
+		var dresserX = Main.interactedDresserTopLeftX;
```
Comment (Chicken-Bones) [inline tModPorter/tModPorter.Tests/TestData/SimpleRenamedVanillaMembersTest.Expected.cs:9]: You could try adding some of these to `Config.cs` yourself! (if you're interested). It doesn't yet have the support for renames across types, but for all the simple stuff that's just renamed within the same type, it should be one line to add!

**Review by Chicken-Bones — APPROVED** [review]
Great stuff, mostly some comments about imports. I'm happy to clean these up as I get to them, but I thought I'd review so you know what to expect for next time.

File: tModPorter/tModPorter.Tests/TestData/GlobalItemTest.Expected.cs (hunk 3 above)
Comment (direwolf420) [reply tModPorter/tModPorter.Tests/TestData/GlobalItemTest.Expected.cs:4]: Rectangle for UseStyle

File: tModPorter/tModPorter.Tests/TestData/FontsTest.Expected.cs (hunk 1 above)
Comment (direwolf420) [reply tModPorter/tModPorter.Tests/TestData/FontsTest.Expected.cs:1]: I'd rather not leave them unless the tester/rewriter breaks because of it

Comment (Chicken-Bones) [reply tModPorter/tModPorter.Tests/TestData/FontsTest.Expected.cs:1]: The test framework will complain that the file produced by tModPorter (which won't remove the using) is different from the Expected.cs

Comment (Chicken-Bones) [reply tModPorter/tModPorter.Tests/TestData/FontsTest.Expected.cs:1]: Anyway, is stuff I can tweak in 20s, so no worries

**Pull request merged**